File: 3127-make-a-square-with-the-same-color/3127-make-a-square-with-the-same-color.py
class Solution:
    def canMakeSquare(self, grid: List[List[str]]) -> bool:
        m,n= len(grid),len(grid[0])
        direction = [[0,0],[1,0],[0,1],[1,1]]
        for i in range(m-1):
            for j in range(n-1):
                black,white=0,0
                for dx,dy in direction:
                    x,y=i+dx,j+dy
                    if grid[x][y]=='B':
                        black+=1
                    else:
                        white+=1
                if black>=3 or white>=3:
                    return True
        return False
                    

File: 3127-make-a-square-with-the-same-color/3127-make-a-square-with-the-same-color.java
class Solution {
    public boolean canMakeSquare(char[][] grid) {
        int n = grid.length;
        int m = grid[0].length;
        for(int i = 0; i < n-1; i++) {
            for(int j = 0; j < m-1; j++) {
                int countBlack = 0;
                int countWhite = 0;
                for(int k = i; k <= i+1; k++) {
                    for(int l = j; l <= j+1; l++) {
                        if(grid[k][l] == 'W') countWhite++;
                        else countBlack++;
                    }
                }
                if(countBlack >= 3 || countWhite >= 3) return true;
            }
        }

        return false;
    }
}